packages/dev/core/src/Meshes/Node/Blocks/radialClonerBlock.ts

```diff
@@ -0,0 +1,285 @@
+        return this._outputs[0];
+    }
+
+    protected _buildBlock(state: NodeGeometryBuildState) {
+        state.executionContext = this;
+
+        this._vertexData = this.geometry.getConnectedValue(state);
+        state.geometryContext = this._vertexData;
+
+        if (!this._vertexData || !this._vertexData.positions || !this.geometry.isConnected) {
+            state.executionContext = null;
+            state.geometryContext = null;
+            this.output._storedValue = null;
+            return;
+        }
+        const count = this.count.getConnectedValue(state) as number;
+
+        if (count <= 0) {
+            this.output._storedValue = this._vertexData;
+            state.executionContext = null;
+            state.geometryContext = null;
+        }
+
+        const origin = this.origin.getConnectedValue(state) as Vector3;
+        const radius = this.radius.getConnectedValue(state) as number;
+        const angleStart: number = this.angleStart.getConnectedValue(state) as number;
+        const angleEnd = this.angleEnd.getConnectedValue(state) as number;
+
+        const clonerCenterOrGlobalForward = this.clonerCenterOrGlobalForward.getConnectedValue(state) as number;
+
+        const transform = this.transform.getConnectedValue(state) as Vector3;
+        const transformPerStepOrTotal = this.transformPerStepOrTotal.getConnectedValue(state) as number;
+
+        const rotation = this.rotation.getConnectedValue(state) as Vector3;
+        const rotationPerStepOrTotal = this.rotationPerStepOrTotal.getConnectedValue(state) as number;
+
+        const scale = this.scale.getConnectedValue(state) as Vector3;
+        const scalePerStepOrTotal = this.scalePerStepOrTotal.getConnectedValue(state) as number;
+
+        const countVector = new Vector3(count - 1, count - 1, count - 1);
+
+        const invertRadians = Math.PI / 180;
+        const angleStartRadians = angleStart * invertRadians;
+        const angleEndRadians = angleEnd * invertRadians;
+        const pieSlice = Math.PI * 2 - (angleStartRadians + angleEndRadians);
+        const rStep = pieSlice / count;
+
+        const transformMatrix = Matrix.Identity();
+        const results = [];
+
+        let rootData: Nullable<VertexData> = null;
+
+        for (this._currentIndex = 0; this._currentIndex < count; this._currentIndex++) {
+            const clone = this._vertexData.clone();
+
+            const centerForward = new Vector3(0, 0, 1);
+            const angle = angleStartRadians + rStep * this._currentIndex;
+            const angleQuat = Quaternion.FromEulerAngles(0, angle, 0);
+            centerForward.rotateByQuaternionToRef(angleQuat, centerForward);
+            const objectOrigin = centerForward.scale(radius).add(origin);
+
+            let transformOffset = transform.clone().scale(this._currentIndex);
+            if (transformPerStepOrTotal == 1) {
+                const transformStep = transform.clone().divide(countVector).scale(this._currentIndex);

```

Comment:
by not doing that we can simplify and accelerate as we have a lot of NonRef operations here

Reply:
Im not sure what you mean here?

Reply:
well a lot of operation generates GC (we need to make sure we are only using ToRef versions of the math functions)